webapi/src/main/scala/org/knora/webapi/slice/admin/repo/service/KnoraProjectRepoLive.scala

```diff
@@ -183,18 +93,96 @@ final case class KnoraProjectRepoLive(
       keywords = keywords.toList.sortBy(_.value),
       logo = logo,
       status = status,
-      selfjoin = selfjoin,
-      ontologies = ontologies.toList
+      selfjoin = selfjoin
     )
 
   override def setProjectRestrictedView(
     project: KnoraProject,
     settings: RestrictedView
   ): Task[Unit] =
-    triplestore.query(Update(Queries.setRestrictedView(project.id, settings.size, settings.watermark)))
+    triplestore.query(Update(ProjectQueries.setProjectRestrictedView(project.id, settings.size, settings.watermark)))
+
+}
+
+object KnoraProjectRepoLive {
 
-  object Queries {
-    def setRestrictedView(projectIri: ProjectIri, size: RestrictedViewSize, watermark: Boolean): String =
+  private object ProjectQueries {
+
+    def findOneByIri(iri: ProjectIri): Construct =
+      Construct(
+        s"""|PREFIX knora-admin: <http://www.knora.org/ontology/knora-admin#>
+            |PREFIX knora-base: <http://www.knora.org/ontology/knora-base#>
+            |PREFIX owl: <http://www.w3.org/2002/07/owl#>
+            |CONSTRUCT {
+            |  ?project ?p ?o .
+            |} WHERE {
+            |  BIND(IRI("${iri.value}") as ?project)
+            |  ?project a knora-admin:knoraProject .
+            |  OPTIONAL {
+            |      ?ontology a owl:Ontology .
+            |      ?ontology knora-base:attachedToProject ?project .
+            |  }
```

Comment:
maybe I'm fundamentally misunderstanding something (or am just tired so late in the evening), but why is it necessary to query for the ontology, if it's no longer added to the `CONSTRUCT` and then used in the `KnoraProject` object?

Reply:
You have it right, removed the needless query parts in f26f56f35447b932d5beb4a56ed263eda4320702